ggml.h

```diff
@@ -231,8 +231,19 @@
 #define GGML_EXIT_SUCCESS 0
 #define GGML_EXIT_ABORTED 1
 
-#define GGUF_MAGIC   0x46554747 // "GGUF"
-#define GGUF_VERSION 2
+#if defined(__linux__) 
+    #include <endian.h>
+    #if BYTE_ORDER == LITTLE_ENDIAN
+        #define GGUF_MAGIC 0x46554747
+    #elif BYTE_ORDER == BIG_ENDIAN
+        #define GGUF_MAGIC 0x47475546
```

Comment:
I think we should either have a comment here to explain it's the same byte sequence in the file or (maybe even better) read raw bytes as Georgi suggested.

Reply:
Now I changed it to char string.